clippy_lints/src/casts/unnecessary_cast.rs

```diff
@@ -59,9 +59,6 @@ pub(super) fn check<'tcx>(
                 lint_unnecessary_cast(cx, expr, literal_str, cast_from, cast_to);
                 return false;
             },
-            LitKind::Int(_, LitIntType::Unsuffixed) | LitKind::Float(_, LitFloatType::Unsuffixed) => {
-                return false;
-            },
```

Comment:
Why are these removed? It looks like that the fix for `clippy_utils/src/numeric_literal.rs` is sufficient to me.

Reply:
It was dead code in the first place

Reply:
To be exact not dead but if the branching is not taken, it will still result to the function returning `false` and since the code triggered afterward is cheap it’s not worth the extra complexity of an early return